webapp/src/components/intersections/ReportsPage.tsx

```diff
@@ -129,6 +129,28 @@ const Page = () => {
   // Usually query is done on backend with indexing solutions
   const paginatedLogs = applyPagination(logs, page, rowsPerPage)
 
+  // Inside the parent component
+  const [selectedReport, setSelectedReport] = useState<ReportMetadata | null>(null)
+  const [isModalOpen, setIsModalOpen] = useState(false)
+
+  const handleViewReport = (report: ReportMetadata) => {
+    setSelectedReport(report)
+    setIsModalOpen(true)
+  }
+
+  const handleCloseReportModal = () => {
+    setIsModalOpen(false)
+    setSelectedReport(null)
+  }
+
+  const handleReportGenerated = () => {
+    setOpenReportGenerationDialog(false)
+    const refreshTime = new Date(Date.now() + 15 * 60 * 1000)
+    toast.success(
+      `Reports usually take 10-15 minutes to generate. Please refresh the page at ${refreshTime.toLocaleTimeString()} to see the new report.`
```

Comment:
If reports really take this long to run should we instead email the finished report to them so they don't need to wait on the website for that long?

Reply:
I think that would make sense - but I see that as part of a larger "add and improve emailing" task - should I add something like that to the backlog?

For clarification, most of the reports are pre-generated, this is a feature for allowing a more specific report to be generated.